ando/tools/generator/nwb2bidsgenerator.py

```diff
@@ -0,0 +1,214 @@
+                        [sb.species, subject_label, sb.sex[0] if sb.sex is not None else None,
+                         sb.date_of_birth, sb.age, sb.genotype, sb.weight]
+            else:
+                subject_label = 'sub-none'
+                if not participants_df['ParticipantID'].str.contains(
+                        subject_label).any():
+                    participants_df.loc[len(participants_df.index)] = \
+                        [None, subject_label, None, None, None, None, None]
+
+            # dataset info:
+            if dataset_desc_json is None:
+                dataset_desc_json = dict(
+                    InstitutionName=nwbfile.institution, InstitutionalDepartmentName=nwbfile.lab,
+                    Name='Electrophysiology', BIDSVersion='1.0.X',
+                    Licence='CC BY 4.0',
+                    Authors=[
+                        list(nwbfile.experimenter) if nwbfile.experimenter is not None else None])
+            # sessions info:
+            subject_path = output_path/subject_label
+            bep_sessions_path = subject_path/f'{subject_label}_sessions.tsv'
+            if not bep_sessions_path.exists():
+                print(f'writing for subject: {subject_label}')
+                sessions_df = pd.DataFrame(
+                    columns=['session_id', '#_trials', 'comment'])
+            else:
+                sessions_df = pd.read_csv(bep_sessions_path, sep='\t')
+            if nwbfile.session_id is not None:
+                session_label = f'ses-{nwbfile.session_id}'
+                # label_count = sessions_df['session_id'].str.contains(session_label).sum()
+                # run_label = f'_run-{label_count}'
+                # if label_count>0:
+            else:
+                session_label = f'ses-{nwbfile.session_start_time.strftime("%Y%m%dT%H%M")}'
+            trials_len = len(
+                nwbfile.trials) if nwbfile.trials is not None else None
+            if not sessions_df['session_id'].str.contains(session_label).any():
+                sessions_count += 1
+                sessions_df.loc[len(sessions_df.index)] = \
+                    [session_label, trials_len, nwbfile.session_description]
+
+            # channels_info:
+            es = [
+                i for i in nwbfile.children if isinstance(
+                    i, ElectricalSeries)]
+            if len(es) > 0:
+                es = es[0]
+                no_channels = es.data.shape[1]
+                sampling_frequency = es.rate
+                conversion = es.conversion
+                unit = es.unit
+                for chan_no in range(no_channels):
+                    channels_df.loc[len(channels_df.index)] = [chan_no, chan_no, 'neural signal',
+                                                               unit,
+                                                               sampling_frequency, conversion]
+
+            # update ephys json:
+            ephys_desc_json = dict(PowerLineFrequency=60.0)
```

Comment:
This might not be true for all datasets. Is there any way we can check the country of data generation to not assume the wrong value here?

Reply:
yeah, Ive added it as an optional arg https://github.com/INT-NIT/AnDO/pull/73/commits/925a4ad9afdcca01f9da8eef3f0cd8c8b3c8a3e1